src/CourseBundle/Repository/CDocumentRepository.php

```diff
@@ -366,6 +366,87 @@ public function findChildNodeByTitle(ResourceNode $parent, string $title): ?Reso
         ;
     }
 
+    /**
+     * Fetches all document data for the given user/group using Doctrine ORM.
+     *
+     * @return CDocument[]
+     */
+    public function getAllDocumentDataByUserAndGroup(
+        Course $course,
+        string $path = '/',
+        int $toGroupId = 0,
+        ?int $toUserId = null,
+        bool $search = false,
+        ?Session $session = null
+    ): array {
+        $qb = $this->createQueryBuilder('d');
+
+        $qb->innerJoin('d.resourceNode', 'rn')
+            ->innerJoin('rn.resourceLinks', 'rl')
+            ->where('rl.course = :course')
+            ->setParameter('course', $course)
+        ;
+
+        // Session filtering
+        if ($session) {
+            $qb->andWhere('(rl.session = :session OR rl.session IS NULL)')
+                ->setParameter('session', $session)
+            ;
+        } else {
+            $qb->andWhere('rl.session IS NULL');
+        }
+
+        // Path filtering - convert document.lib.php logic to Doctrine
+        if ('/' !== $path) {
```

Comment:
This looks like the old files strategy of looking at file paths to define hierarchies. This is not the way this should be done in Chamilo 2.0. Rather, you should use ResourceFile and ResourceNode entities for that (ask if you need more details)